tools/misc/tpm2_encodeobject.c

```diff
@@ -182,10 +182,10 @@ encode(void)
     }
 
     tpk->type = OBJ_txt2obj(OID_loadableKey, 1);
-    tpk->parent = ASN1_INTEGER_new();
-    tpk->privkey = ASN1_OCTET_STRING_new();
-    tpk->pubkey = ASN1_OCTET_STRING_new();
-    if (!tpk->type || !tpk->privkey || !tpk->pubkey || !tpk->parent) {
+    //tpk->parent = ASN1_INTEGER_new();
+    //tpk->privkey = ASN1_OCTET_STRING_new();
+    //tpk->pubkey = ASN1_OCTET_STRING_new();
+    if (!tpk->type /*|| !tpk->privkey || !tpk->pubkey || !tpk->parent*/) {
```

Comment:
@idesai @williamcroberts I am not sure if these are required here, but I think this is one spot where memory is allocated and not necessarily used. Commenting these out gets rid of the memory leaks in encodeobject.c for now. Please let me know what you guys think?